data-processor/src/main/java/io/micronaut/data/processor/visitors/RepositoryTypeElementVisitor.java

```diff
@@ -587,4 +593,46 @@ private SourcePersistentEntity resolveEntityForCurrentClass() {
         return null;
     }
 
+    /**
+     * Annotates method element with {@link EntityRepresentation} if an entity is marked with it.
+     *
+     * @param methodElement the method element
+     * @param genericReturnType the generic return type for the method
+     * @param entity the source persistent entity
+     */
+    private void annotateEntityRepresentationIfPresent(MethodElement methodElement, ClassElement genericReturnType, SourcePersistentEntity entity) {
+        AnnotationValue<EntityRepresentation> entityRepresentationAnnotationValue = entity.getAnnotation(EntityRepresentation.class);
+        if (entityRepresentationAnnotationValue != null) {
+            methodElement.annotate(entityRepresentationAnnotationValue);
+            ClassElement entityClassElement = entity.getClassElement();
```

Comment:
Do we need to do this? AnnotationMetadata for a method should include class annotations

Reply:
If annotation is not on repository but on an entity, then I think we need it?

Reply:
Done.